frontend/app/actions/ActionRefiners.scala

```diff
@@ -91,8 +91,10 @@ object ActionRefiners extends LazyLogging {
 
   def checkTierChangeTo(targetTier: PaidTier) = new ActionFilter[SubReqWithSub] {
     override protected def filter[A](request: SubReqWithSub[A]): Future[Option[Result]] = {
-      if (!request.touchpointBackend.memberService.subscriptionUpgradableTo(request.subscriber.subscription, targetTier)) {
-        Future.successful(Some(Ok(views.html.tier.upgrade.unavailable(request.subscriber.subscription.plan.tier, targetTier))))
+      if (!request.touchpointBackend.memberService.upgradeTierIsValidForCurrency(request.subscriber.subscription, targetTier)) {
+        Future.successful(Some(Ok(views.html.tier.upgrade.unavailableTierForCurrency(request.subscriber.subscription.plan.tier, targetTier))))
```

Comment:
You can extract the `Future.successful()` outside of this `if` expression to reduce repetition.